firedrake/mesh.py

```diff
@@ -1032,6 +1032,27 @@ def coordinates(self, value):
 
         raise AttributeError(message)
 
+    @utils.cached_property
+    def cell_sizes(self):
+        """A :class`~.Function` in the :math:`P^1` space containing the local mesh size.
+
+        This is computed by the :math:`L^2` projection of the local mesh element size."""
```

Comment:
Thanks, this now clarifies the P1 mesh size, and why that needs to pre-calculated like `cell_orientations`...

Reply:
Yes, I'll add some explanatory comments on the tsfc side.